### PR TITLE
implement changes so that users can confirm the pedelec reservation

### PR DESCRIPTION
**Pull request by CapChrisCap**

### CODE REVIEW

File: PedelecApp/src/de/tum/in/www1/PedelecApp.java [hunk 1]
```diff
@@ -6,60 +6,74 @@
...
-		});
+
+        reserveButton.setOnAction(new EventHandler<ActionEvent>() {
+            @Override
+            public void handle(ActionEvent e) {
+                System.out.println("Reserve Button clicked");
+
+                LocalDate XX1 = datePicker.getValue();
+
+                String x_1 = timeTextField.getText(), x_2 = x_1, x_3;
```
Comment (reichmann92) [inline PedelecApp/src/de/tum/in/www1/PedelecApp.java]: use speaking variables - what dows x_1 mean?

File: PedelecApp/src/de/tum/in/www1/PedelecApp.java [hunk 2]
```diff
@@ -6,60 +6,74 @@
...
+                ButtonType x__4 = new Alert(Alert.AlertType.CONFIRMATION, "" + x_3).showAndWait().get();
+
+                if (x__4 == ButtonType.OK) {
+                    Reservation r33 = new Reservation(), r34;
+                    r33.setBike(x_4);
+                    r33.setStartDate(XX1);
+                    r33.setStartTime(timeTextField.getText().concat(""));
+                    r34 = r33;
+                    r34.save();
+                    System.out.println("Reservation confirmed");
```
Comment (reichmann92) [inline PedelecApp/src/de/tum/in/www1/PedelecApp.java]: No comments - please comment your code

File: PedelecApp/src/de/tum/in/www1/PedelecApp.java [hunk 3]
```diff
@@ -6,60 +6,74 @@
...
-		        reservation.save();
-		    }
-		});
+
+        reserveButton.setOnAction(new EventHandler<ActionEvent>() {
+            @Override
+            public void handle(ActionEvent e) {
+                System.out.println("Reserve Button clicked");
+
+                LocalDate XX1 = datePicker.getValue();
```
Comment (reichmann92) [inline PedelecApp/src/de/tum/in/www1/PedelecApp.java]: what does XX1 mean??

File: PedelecApp/src/de/tum/in/www1/PedelecApp.java [hunk 4]
```diff
@@ -6,60 +6,74 @@
...
+
+                String x_1 = timeTextField.getText(), x_2 = x_1, x_3;
+                x_2 = pedelecNametext.getText();
+                String x_4 = x_2;
+                x_3 = "Please confirm your reservation of " + x_2 + " at " + XX1.format(DateTimeFormatter.ISO_LOCAL_DATE) + " " + timeTextField.getText();
+                ButtonType x__4 = new Alert(Alert.AlertType.CONFIRMATION, "" + x_3).showAndWait().get();
+
+                if (x__4 == ButtonType.OK) {
+                    Reservation r33 = new Reservation(), r34;
```
Comment (reichmann92) [inline PedelecApp/src/de/tum/in/www1/PedelecApp.java]: what mean reservation r33

File: PedelecApp/src/de/tum/in/www1/PedelecApp.java [hunk 5]
```diff
@@ -6,60 +6,74 @@
...
+        reserveButton.setOnAction(new EventHandler<ActionEvent>() {
+            @Override
+            public void handle(ActionEvent e) {
+                System.out.println("Reserve Button clicked");
+
+                LocalDate XX1 = datePicker.getValue();
+
+                String x_1 = timeTextField.getText(), x_2 = x_1, x_3;
+                x_2 = pedelecNametext.getText();
+                String x_4 = x_2;
```
Comment (reichmann92) [inline PedelecApp/src/de/tum/in/www1/PedelecApp.java]: really required?

**Review by reichmann92 — CHANGES_REQUESTED** [review]
Please fix

File: PedelecApp/src/de/tum/in/www1/PedelecApp.java (hunk 1 above)
Comment (CapChrisCap) [reply PedelecApp/src/de/tum/in/www1/PedelecApp.java]: Fixed and changes to datePickerValue

File: PedelecApp/src/de/tum/in/www1/PedelecApp.java (hunk 2 above)
Comment (CapChrisCap) [reply PedelecApp/src/de/tum/in/www1/PedelecApp.java]: Now the code is so clean that it is not necessairy to add comments

File: PedelecApp/src/de/tum/in/www1/PedelecApp.java (hunk 3 above)
Comment (CapChrisCap) [reply PedelecApp/src/de/tum/in/www1/PedelecApp.java]: Renamed to datePickerValue

File: PedelecApp/src/de/tum/in/www1/PedelecApp.java (hunk 5 above)
Comment (CapChrisCap) [reply PedelecApp/src/de/tum/in/www1/PedelecApp.java]: No, removed

File: PedelecApp/src/de/tum/in/www1/PedelecApp.java (hunk 4 above)
Comment (CapChrisCap) [reply PedelecApp/src/de/tum/in/www1/PedelecApp.java]: Renamed to reservation

**Comment by CapChrisCap** [comment]
Updated the changes, please review again

**Review by reichmann92 — APPROVED** [review]

**Comment by reichmann92** [comment]
Your Code is beautiful

**Pull request merged**